static/app/components/smartSearchBar/index.tsx

```diff
@@ -242,7 +242,7 @@ type Props = WithRouterProps & {
    * Called when the user has closed the search dropdown.
    * Occurs on escape, tab, or clicking outside the component.
    */
-  onClose?: (value: string) => void;
+  onClose?: (value: string, validSearch: boolean) => void;
```

Comment:
Coming in hot with more suggestions (sorry 😄)

Can we make the second parameter an object? like: 

```ts
(value: string, info: { validSearch: boolean }) => void;
```

I'm anticipating that we may need to add more properties to this handler and that will make doing so easiser.

Reply:
Yup, can do! Updated!